### DISCUSSION TITLE
compile problem

### DISCUSSION THREAD

**Discussion by Rudo909** [rfquack/RFQuack#49] [Q&A]
Hello,

I like to test RFQuack, but i'm a newbe with VSEditor and PlatformIO. 
After i  cloned the GIT project i get this fault while compiling.

src/radio/RadioLibWrapper.h:18:10: fatal error: RadioLib.h: No such file or directory.

main.cpp:
cannot open source file "RadioLib.h" (dependency of "C:\VSCode\RFQuack\radio\drivers.h")

Please can anyone help me find out which step I am overlooking? I alreday tried to install the Radioib library manualy

Thank you in Advance.

**Comment by phretor** [comment]
Hello! Could you please paste the complete command line and complete output?

**Note:** please also check the official docs at https://rfquack.org in case you haven't done so already.

**Comment by Rudo909** [comment]
Goodmorning Federico,

this is the complete output from VSCode on Windows10 without using command line. I have red your documenation by the way :-)

 *  Executing task: C:\Users\rudo\.platformio\penv\Scripts\platformio.exe run 

Processing featheresp32 (platform: platformio/espressif32@^5.2.0; board: featheresp32; framework: arduino)
-------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------Verbose mode can be enabled via `-v, --verbose` option
j2 -f env C:\Users\rudo\Documents\VSCode\RFQuack\docker\project\src\main.cpp.j2 C:\Users\rudo\Documents\VSCode\RFQuack\build.env > C:\Users\rudo\Documents\VSCode\RFQuack\src\main.cpp
CONFIGURATION: https://docs.platformio.org/page/boards/espressif32/featheresp32.html
PLATFORM: Espressif 32 (5.4.0) > Adafruit ESP32 Feather
HARDWARE: ESP32 240MHz, 320KB RAM, 4MB Flash
DEBUG: Current (cmsis-dap) External (cmsis-dap, esp-bridge, esp-prog, iot-bus-jtag, jlink, minimodule, olimex-arm-usb-ocd, olimex-arm-usb-ocd-h, olimex-arm-usb-tiny-h, olimex-jtag-tiny, tumpa)
PACKAGES:
 - framework-arduinoespressif32 @ 3.20006.221224 (2.0.6)
 - tool-esptoolpy @ 1.40400.0 (4.4.0)
 - toolchain-xtensa-esp32 @ 8.4.0+2021r2-patch5
LDF: Library Dependency Finder -> https://bit.ly/configure-pio-ldf
LDF Modes: Finder ~ chain, Compatibility ~ soft
Installing Protocol Buffers dependencies
Requirement already satisfied: protobuf>=3.19.1 in c:\users\rudo\.platformio\penv\lib\site-packages (4.25.2)
[nanopb] Skipping 'src\rfquack.proto' (rfquack.options)
Found 40 compatible libraries
Scanning dependencies...
Dependency Graph
|-- Nanopb @ 0.4.6+4
|-- TinyGSM @ 0.6.2
|-- MQTT @ 2.4.7
|-- Queue @ 1.6.0
|-- ArduinoLog @ 1.0.4
|-- base64 @ 1.4.0
|-- WiFi @ 2.0.0
|-- WiFiClientSecure @ 2.0.0
Building in release mode
Compiling .pio\build\featheresp32\src\main.cpp.o
Archiving .pio\build\featheresp32\liba32\libNanopb.a
In file included from src/radio/RFQCC1101.h:4,
                 from src/radio/drivers.h:4,
                 from src/main.cpp:67:
src/radio/RadioLibWrapper.h:18:10: fatal error: RadioLib.h: No such file or directory

******************************************************************
* Looking for RadioLib.h dependency? Check our library registry!
*
* CLI  > platformio lib search "header:RadioLib.h"
* Web  > https://registry.platformio.org/search?q=header:RadioLib.h
*
******************************************************************

 #include <RadioLib.h>
          ^~~~~~~~~~~~
compilation terminated.
Compiling .pio\build\featheresp32\lib001\MQTT\lwmqtt\string.c.o
Compiling .pio\build\featheresp32\lib6b6\Queue\cppQueue.cpp.o
Compiling .pio\build\featheresp32\libc14\ArduinoLog\ArduinoLog.cpp.o
Compiling .pio\build\featheresp32\lib422\WiFi\WiFi.cpp.o
Compiling .pio\build\featheresp32\lib422\WiFi\WiFiAP.cpp.o
Compiling .pio\build\featheresp32\lib422\WiFi\WiFiClient.cpp.o
Compiling .pio\build\featheresp32\lib422\WiFi\WiFiGeneric.cpp.o
Compiling .pio\build\featheresp32\lib422\WiFi\WiFiMulti.cpp.o
Compiling .pio\build\featheresp32\lib422\WiFi\WiFiSTA.cpp.o
Compiling .pio\build\featheresp32\lib422\WiFi\WiFiScan.cpp.o
*** [.pio\build\featheresp32\src\main.cpp.o] Error 1
======================================================================================== [FAILED] Took 15.21 seconds ========================================================================================
Environment    Status    Duration
-------------  --------  ------------
featheresp32   FAILED    00:00:15.212
=================================================================================== 1 failed, 0 succeeded in 00:00:15.212 ===================================================================================
 *  The terminal process "C:\Users\rudo\.platformio\penv\Scripts\platformio.exe 'run'" terminated with exit code: 1. 
 *  Terminal will be reused by tasks, press any key to close it. 




I

**Reply by phretor** [reply]
We can't know what VSCode decides to install and run out of the box so it's difficult to reproduce the issue you're experiencing.

Please refer to the compile instructions https://rfquack.org/#quickstart and let us know if those yields any errors.

**Comment by Rudo909** [comment]
Thank you for time Federico.  
I have managed to compile from the cammand line on a Ubuntu machine. Unfortunally i get some fault while compiling. I have tried a Sparkfun32 Things Plus with one CC1101 and  an ECRFv2.  (the files included is with the ECRFv2)
I can connect with the dongles but it seems that the tranceivers are not functioning/reacting. 

Please can anyone help me find out which step I am overlooking?

Thank you in Advance.


 
[build.env.txt](https://github.com/rfquack/RFQuack/files/13948059/build.env.txt)
[log.txt](https://github.com/rfquack/RFQuack/files/13948060/log.txt)
[platformio.ini.txt](https://github.com/rfquack/RFQuack/files/13948061/platformio.ini.txt)
